trezor/trezor-firmware
# Support extendable backup flag in SLIP-39

Related to:
- SLIP-39 spec update: https://github.com/satoshilabs/slips/commit/8d060706b549af6443e04f55605b71f65c981663
- Reference implementation update: https://github.com/trezor/python-shamir-mnemonic/commit/1525df19df504b1f69b49179140119959f317f24
- Feature: https://github.com/trezor/trezor-firmware/issues/3640
- Feature implementation: https://github.com/trezor/trezor-firmware/pull/3743

TODO:
- [x] Update `shamir-mnemonic` package once version 0.3.0 is released.
- [x] Update screens.
- [x] Maybe add some device/upgrade tests based on the QA list below.

Notes to @trezor/qa:
- Thoroughly test Shamir backup.
- Test case 1:
    1. Install old firmware (2.7.0 or earlier).
    2. Reset device and choose Shamir backup, but do not back up.
    3. Write down some address or XPUB from passphrase-less wallet.
    4. Write down some address or XPUB from passphrased wallet.
    5. Install new firmware.
    6. Ensure that the addresses of the wallets remain the same.
    7. Create a backup and reboot.
    8. Ensure that the addresses of the wallets remain the same.
    9. Wipe the device and recover using the backup.
    10. Ensure that the addresses of the wallets remain the same.
- Test case 2:
    1. Install old firmware.
    2. Reset device and choose Shamir backup and create a backup.
    3. Write down some address or XPUB from passphrase-less wallet.
    4. Write down some address or XPUB from passphrased wallet.
    5. Install new firmware.
    6. Ensure that the addresses of the wallets remain the same.
    7. Wipe the device and recover using the backup.
    8. Ensure that the addresses of the wallets remain the same.
- TC3: Ensure that Shamir backups created with **_new_** firmware can be successfully recovered on **_new_** firmware, i.e. check that addresses remain the same.
- TC4: Ensure that Shamir backups created with **_new_** firmware are not accepted by **_old_** firmware.
